index.js

```diff
@@ -3,37 +3,37 @@
  * Author: Theekshana Wijesinghe
  */
 class StringValidator {
-	constructor(value) {
-		if (value && typeof value !== 'string') throw new Error('Invalid string type');
+	constructor(value = null) {
+		if (value !== null && typeof value !== 'string') throw new Error('Invalid string type');
```

Comment:
@suthagar23 shall we allow, `null/undefined` here.
My argument is as follows, 
suppose we want to validate a field inside an object `obj.someField`, it is possible that this can be `undefined`,
and a user has the following validation statement.
```
new StringValidator(Obk.someField).max(50).min(2).allow(undefined).validate();
```
Given that default value `null` for the constructor, we are now making above statement `false`.
I suggest following, let's define a method to allow only following to constructor given it's been called with a `falsy` value
``` 
const isAllowed = (value) => value === undefined || value === null 
```
this method can be used in each method as well.

Reply:
A point on the above, empty string `''` is also a valid candidate to be allowed.

Reply:
Thanks for the explanation @theekshanawj. As we only have to align for `allow` method to accept `undefined / null / ''`, and other utility methods won't worry about this. 

So I created a class variable called `isNullOrUndefined` to check and keep a boolean for empty checks. So all other utility methods can use that variable to move forward which might reduce the time for empty checks in chained methods.

Reply:
:+1: awesome job @suthagar23